Formula/whois.rb

```diff
@@ -19,15 +19,23 @@ class Whois < Formula
   depends_on "libidn2"
 
   def install
-    ENV.append "LDFLAGS", "-L/usr/lib -liconv"
+    on_macos do
+      ENV.append "LDFLAGS", "-L/usr/lib -liconv"
+    end
 
-    system "make", "whois", "HAVE_ICONV=1"
+    have_iconv = "HAVE_ICONV=1"
+
+    on_linux do
+      have_iconv = "HAVE_ICONV=0"
+    end
+
+    system "make", "whois", have_iconv
     bin.install "whois"
     man1.install "whois.1"
     man5.install "whois.conf.5"
   end
 
   test do
-    system "#{bin}/whois", "brew.sh"
+    system "#{bin}/whois", "brew.sh" if Pathname.new("/etc/services").exist?
```

Comment:
Can't we do another test on linux? Or create our own config file?

Reply:
Actually this file seems to exist Linux CI and the test works fine there so I think this is unneeded and instead I'm going to remove the change to the test in linuxbrew-core.